common/modules/farm/custom_token_rewards/src/custom_token_rewards.rs

```diff
@@ -0,0 +1,139 @@
+#![no_std]
+
+elrond_wasm::imports!();
+elrond_wasm::derive_imports!();
+
+use common_structs::Nonce;
+
+use common_errors::{ERROR_BAD_INPUT_TOKEN, ERROR_NOT_AN_ESDT};
+
+#[elrond_wasm::module]
+pub trait CustomTokenRewardsModule:
+    config::ConfigModule
+    + farm_token::FarmTokenModule
+    + token_send::TokenSendModule
+    + pausable::PausableModule
+    + admin_whitelist::AdminWhitelistModule
+    + elrond_wasm_modules::default_issue_callbacks::DefaultIssueCallbacksModule
+{
+    fn calculate_per_block_custom_rewards(
+        &self,
+        current_block_nonce: Nonce,
+        last_reward_block_nonce: Nonce,
+    ) -> BigUint {
+        let final_reward_block = self.custom_token_final_reward_block().get();
+        if current_block_nonce <= last_reward_block_nonce
+            || final_reward_block < current_block_nonce
+            || !self.produces_per_block_custom_rewards()
+        {
+            return BigUint::zero();
+        }
+
+        let mut rewards = BigUint::zero();
+        if final_reward_block == current_block_nonce {
```

Comment:
```suggestion
        if current_block_nonce >= final_reward_block {
```
This condition would be skipped if nobody would interact with the contract in the `final_reward_block`.